### PR TITLE
Be able to configure the prompt

### PR DESCRIPTION
**Pull request by dennisdegreef**
We are using psysh for multiple applications, and would like to have the prompt configurable.

```
dennisdegreef@appname> 
```

This change should allow it.

### CODE REVIEW

**Comment by dennisdegreef** [comment]
The build fails on something that seems unrelated to my change

```
$ phpenv global 7.1 2>/dev/null

7.1 is not pre-installed; installing

Downloading archive: https://s3.amazonaws.com/travis-php-archives/binaries/ubuntu/12.04/x86_64/7.1-php.tar.bz2
$ curl -s -o archive.tar.bz2 $archive_url && tar xjf archive.tar.bz2 --directory /
bzip2: (stdin) is not a bzip2 file.
tar: Child returned status 2
tar: Error is not recoverable: exiting now
$ phpenv global 7.1
rbenv: version `7.1' not installed
```

**Comment by dennisdegreef** [comment]
Let me know if there is anything I can do to help get this merged. Thanks!

File: src/Psy/Shell.php [hunk 1]
```diff
@@ -866,7 +867,15 @@ protected function hasCommand($input)
      */
     protected function getPrompt()
     {
-        return $this->hasCode() ? static::BUFF_PROMPT : static::PROMPT;
+        if ($this->hasCode()) {
+            return static::BUFF_PROMPT;
+        }
+
+        if ($this->config->getPrompt() !== null) {
```
Comment (bobthecow) [inline src/Psy/Shell.php]:
```
return $this-config->getPrompt() ?: static::PROMPT;
```

**Comment by bobthecow** [comment]
Thank you!

**Pull request merged**